### PR TITLE
Add documentation and specs for stubbing out options requests

### PR DESCRIPTION
**Pull request by rimian**
I was having trouble stubbing out OPTIONS requests. Here are some specs that prove it works.

### CODE REVIEW

File: spec/lib/proxy_spec.rb [hunk 1]
```diff
@@ -22,6 +22,10 @@
   it 'should proxy DELETE requests' do
     expect(http.delete('/echo').body).to eql 'DELETE /echo'
   end
+
+  it 'should proxy OPTIONS requests' do
+    expect(http.run_request(:options, '/echo', nil, nil).body).to eql 'OPTIONS /echo'
+  end
```
Comment (rimian) [inline spec/lib/proxy_spec.rb]: It collides with Faraday #options here. This work around does the trick.

**Comment by ronwsmith** [comment]
Thanks for the contribution!  Mind adding something to the README too?

**Comment by rimian** [comment]
@ronwsmith done! :)

**Comment by ronwsmith** [comment]
Included in [v0.7.0](https://rubygems.org/gems/puffing-billy/versions/0.7.0)

**Pull request merged**